app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeResultContract.kt

```diff
@@ -0,0 +1,34 @@
+/* Copyright 2023 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.compose
+
+import android.content.Context
+import android.content.Intent
+import androidx.activity.result.contract.ActivityResultContract
+
+class ComposeResultContract : ActivityResultContract<ComposeActivity.ComposeOptions, String>() {
```

Comment:
Technical curiosity: Does this need a contract?

(The only other two usages I see are "take picture" and "login with browser" - so external stuff.)

Reply:
It looked to me like it did? As in, android studio yelled at me that the old `startActivityForResult` was deprecated, and the docs led me to `ActivityResultContract` 🤷